File: frontend/src/types/invitation.ts
export enum InvitationStatus {
    PENDING = 'PENDING',      // Oczekuje na akceptację
    ACCEPTED = 'ACCEPTED',    // Zaakceptowane - aktywna współpraca
    EXPIRED = 'EXPIRED',      // Wygasłe przez timeout (7 dni)
    ENDED = 'ENDED'           // Współpraca zakończona celowo (przez usera lub trenera)
}

export interface Invitation {
    id: string;
    trainerId: string;
    clientEmail: string;
    clientId?: string;      // ID klienta (null dla PENDING, wypełniane przy ACCEPTED)
    code: string;
    status: InvitationStatus;
    createdAt: number;
    expiresAt: number;
}

export interface InvitationRequest {
    email: string;
}

export interface InvitationResponse extends Invitation {}

export interface AcceptInvitationRequest {
    code: string;
}

export interface MessageResponse {
    message: string;
}

File: backend/src/main/java/com/noisevisionsoftware/vitema/model/invitation/Invitation.java
package com.noisevisionsoftware.vitema.model.invitation;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Invitation {
    private String id;
    private String trainerId;
    private String clientEmail;
    private String code;
    private InvitationStatus status;
    private Long createdAt;
    private Long expiresAt;
}
